mesonbuild/compilers/detect.py

```diff
@@ -983,7 +984,7 @@ def detect_rust_compiler(env: 'Environment', for_machine: MachineChoice) -> Rust
                     extra_args['machine'] = cc.linker.machine
                 else:
                     exelist = cc.linker.exelist + cc.linker.get_always_args()
-                    if 'ccache' in exelist[0]:
+                    if 'ccache' in os.path.basename(exelist[0]) and len(exelist) > 1:
```

Comment:
not sure about `len(exelist) > 1` when `exelist` is empty after the remove something is clearly wrong. Report an error with the context?

Reply:
Maybe an exact check for cache/cache.exe would be better?

Reply:
If the user specifies that `CC=ccache` then something is indeed quite wrong. However, it will already fail with e.g. `ERROR: Unknown compiler(s): [['/usr/bin/ccache']]` when detecting the C compiler, which happens before rust.